modules/videoio/src/cap_avfoundation_mac.mm

```diff
@@ -293,7 +293,7 @@ - (IplImage*)getOutput;
 
     [mCaptureSession release];
     [mCaptureDeviceInput release];
-    [mCaptureDevice release];
+    // [mCaptureDevice release]; fix #7833
```

Comment:
Hi @alalek, I tried to do as you suggested  (mCaptureDevice = nil;) and it leads to internal opencv crush although it is different in this case. Also cap_avfoundation.mm doesn't have [mCaptureDevice release] and I assume it works well as is, without any changes. 
So may be it makes sense to merge as is because additional changes are causing problems.